packages/openapi-fetch/test/common/response.test.ts

```diff
@@ -20,22 +20,38 @@ describe("response", () => {
       // 2. assert data is not undefined inside condition block
       if (result.data) {
         assertType<NonNullable<Resource[]>>(result.data);
-        assertType<undefined>(result.error);
+        // @ts-expect-error FIXME: This is a limitation within Typescript
+        assertType<never>(result.error);
       }
       // 2b. inverse should work, too
       if (!result.error) {
         assertType<NonNullable<Resource[]>>(result.data);
-        assertType<undefined>(result.error);
+        assertType<never>(result.error);
+      }
+
+      if (result.status === 200) {
+        assertType<NonNullable<Resource[]>>(result.data);
+        assertType<never>(result.error);
+      }
+
+      if (result.status === 500) {
+        assertType<never>(result.data);
+        assertType<Error>(result.error);
+      }
+
+      // @ts-expect-error 204 is not defined in the schema
+      if (result.status === 204) {
```

Comment:
This is a REALLY good test I want to keep—testing what happens when a status code is impossible for a given endpoint! But we should test that `204` works somewhere, even if it’s a different test / different endpoint (doesn’t matter where, as long as it’s somewhere, but soft recommendation to just add `204` here since we have all the other checks, and test for the “missing” case in its own test).